### PR TITLE
Missing stacktrace basic reproduction test

### PR DESCRIPTION
**Pull request by davidmartos96**
@isoos Here is a basic test case which reproduces #13 . I've included the docker_process package, so it will run a local postgres container.

### CODE REVIEW

File: test/docker.dart [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import 'dart:io';
+
+import 'package:docker_process/containers/postgres.dart';
+import 'package:test/test.dart';
+
+const _kContainerName = 'postgres-dart-test';
+
+Future<void> startPostgresContainer() async {
+  setUpAll(() async {
```
Comment (isoos) [inline test/docker.dart:9]: let's also add a teardownAll that stops the container

File: test/docker.dart [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import 'dart:io';
+
+import 'package:docker_process/containers/postgres.dart';
+import 'package:test/test.dart';
+
+const _kContainerName = 'postgres-dart-test';
+
+Future<void> startPostgresContainer() async {
```
Comment (isoos) [inline test/docker.dart:8]: I think async happens only in the setupAll, so this method should rather be something like `void usePostgresDocker() {`.

File: test/docker.dart [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      pgPort: 5432,
+      pgDatabase: 'dart_test',
+      pgUser: 'dart',
+      pgPassword: 'dart',
+      cleanup: true,
+    );
+  });
+}
+
+Future<bool> isPostgresContainerRunning() async {
```
Comment (isoos) [inline test/docker.dart:32]: Let's make this private.

File: test/docker.dart [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      return;
+    }
+
+    final isRunning = await isPostgresContainerRunning();
+    if (isRunning) {
+      return;
+    }
+
+    await startPostgres(
+      name: _kContainerName,
```
Comment (isoos) [inline test/docker.dart:21]: This is a nice thing to have here, I haven't really thought that you'd match the CI's config, but it makes sense.

File: test/error_handling_test.dart [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import 'package:postgres/postgres.dart';
+import 'package:test/test.dart';
+
+import 'docker.dart';
+
+Future<void> main() async {
```
Comment (isoos) [inline test/error_handling_test.dart:6]: Test main can be `void main() {` without the `async` block.

**Review by isoos — COMMENTED** [review]
Looks good, mostly nitpicking requests.

**Comment by davidmartos96** [comment]
@isoos I've updated the PR with the comments

**Comment by isoos** [comment]
this looks good and it is a good baseline for debugging... can I just merge it?

**Comment by davidmartos96** [comment]
> this looks good and it is a good baseline for debugging... can I just merge it?

Yes, go ahead! I've tried debugging it, but I didn't know where to start because of the event system the library uses.

**Comment by isoos** [comment]
> I've tried debugging it, but I didn't know where to start because of the event system the library uses.

There are a few places where one can start putting e.g. debug/print statements:
- just before we call any `Completer.complete` or `Completer.completeError`
- when the new data coming in is being parsed.

Feel free to update the issue if you find something, I'll try to find time to debug this over the weekend too...

**Pull request merged**